pkg/server/config.go

```diff
@@ -42,6 +61,36 @@ func (cfg *HandlerConfig) Validate() error {
 	return nil
 }
 
+// ValidateMappingConfig validates the handler config for mapping service after load.
+func (cfg *HandlerConfig) ValidateMappingConfig() error {
+	if err := cfg.Validate(); err != nil {
+		return err
+	}
+
+	// For mapping server, we also require a failure topic ID.
+	if cfg.FailureTopicID == "" {
+		return fmt.Errorf("PMAP_FAILURE_TOPIC_ID is empty and require a value for mapping service")
+	}
+
+	if cfg.MappingConfig.DefaultResourceScope == "" {
+		return fmt.Errorf(`PMAP_RESOURCE_SCOPE is empty and require a value from one of the following format:\n
+			projects/{PROJECT_ID}\n
```

Comment:
Your way of using literal string is problematic. Your output will include unnecessary tabs.

```suggestion
		allowedScopes := [
				"projects/[PROJECT_ID]",
				"projects/[PROJECT_NUMBER]",
				"folders/[FOLDER_NUMBER]",
				"organizations/[ORGNANIZATION_NUMBER]",
		]
		return fmt.Errorf("PMAP_MAPPING_RESOURCE_SCOPE is required in one of the formats: %v", allowedScopes)
```

Reply:
Define `allowedScopes` as a var on the top

Reply:
If we agree to use map (see reason from the other conversation), I think we can keep allowedScopes in the validate function.

Reply:
See my other comment. And again, your usage of ` is wrong. See the difference here https://go.dev/play/p/Ag73GyblH4l